### PR TITLE
centralisation, optimisation and streamlining of some of InputHandler functionality

### PR DESCRIPTION
**Pull request by TAImatem**

### CODE REVIEW

File: OWML.ModHelper.Input/ModInputHandler.cs [hunk 1]
```diff
@@ -285,14 +279,30 @@ private List<string> GetCollisions(ReadOnlyCollection<long> hashes)
...
+        public List<string> GetCollisions(string combinations)
+        {
+            var hashes = new List<long>();
+            foreach (var combo in combinations.Split('/'))
+            {
+                var hash = ModInputLibrary.StringToHash(combo);
+                if (hash <= 0)
+                {
+                    var toReturn = new List<string>();
+                    toReturn.Add(((RegistrationCode)(-hash)).ToString());
```
Comment (amazingalek) [inline OWML.ModHelper.Input/ModInputHandler.cs:299]: not used

**Comment by amazingalek** [comment]
Could the input library be a normal class/object, injected into the constructors where it's needed, instead of being static?

File: OWML.ModHelper.Input/ModInputHandler.cs (hunk 1 above)
Comment (TAImatem) [reply OWML.ModHelper.Input/ModInputHandler.cs:299]: lol, I forgot to write 'return toReturn;'

Comment (TAImatem) [reply OWML.ModHelper.Input/ModInputHandler.cs]: figured might as well turn it into one-liner

**Comment by TAImatem** [comment]
you want me to make texture-related stuff non-static?

**Comment by amazingalek** [comment]
> 
> 
> you want me to make texture-related stuff non-static?
> Technically I can pass ModInputLibrary around as InputHandler's field

yeah, what do you think?

File: OWML.ModHelper.Input/ModInputHandler.cs [hunk 2]
```diff
@@ -24,16 +20,22 @@ public class ModInputHandler : IModInputHandler
...
         private readonly IModLogger _logger;
         private readonly IModConsole _console;
 
+        public IModInputTextures Textures { get; }
+
         public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher, IOwmlConfig owmlConfig, IModEvents events)
         {
+            var textures = new ModInputTextures();
+            textures.FillTextureLibrary();
+            Textures = textures;
```
Comment (amazingalek) [inline OWML.ModHelper.Input/ModInputHandler.cs:37]: You could save a line:
```
Textures = new ModInputTextures();
Textures.FillTextureLibrary();
```

File: OWML.ModHelper.Input/ModInputHandler.cs [hunk 3]
```diff
@@ -49,34 +51,31 @@ public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper pa
...
         {
             UpdateCurrentCombination();
-            var intKey = (int)code;
-            if (intKey >= MaxUsefulKey)
-            {
-                intKey -= (intKey - MaxUsefulKey + GamePadKeyDiff) / GamePadKeyDiff * GamePadKeyDiff;
-            }
-            return UnityEngine.Input.GetKey(code) && _currentCombination != null && Time.realtimeSinceStartup - _timeout[intKey] < Cooldown;
+            var cleanKey = ModInputLibrary.NormalizeKeyCode(key);
+            return UnityEngine.Input.GetKey(cleanKey) && _currentCombination != null && Time.realtimeSinceStartup - _timeout[(int)cleanKey] < Cooldown;
```
Comment (amazingalek) [inline OWML.ModHelper.Input/ModInputHandler.cs:58]: Could have newline after each `&&`

File: OWML.ModHelper.Input/ModInputHandler.cs (hunk 2 above)
Comment (TAImatem) [reply OWML.ModHelper.Input/ModInputHandler.cs:37]: FillTecctureLibrary is internal, I preferred that to cast

File: OWML.ModHelper.Input/ModInputHandler.cs (hunk 3 above)
Comment (amazingalek) [reply OWML.ModHelper.Input/ModInputHandler.cs:58]: meh 🤷

**Comment by amazingalek** [comment]
Looks great! :)
Haven't tested it tho... :P

File: OWML.ModHelper.Input/ModInputHandler.cs (hunk 3 above)
Comment (TAImatem) [reply OWML.ModHelper.Input/ModInputHandler.cs]: done

**Comment by TAImatem** [comment]
Well, I tested it and it's actually more stable (i.e. ModComboInput won't crash if alt is written instead of lalt)

**Pull request merged**